d2layouts/d2near/layout.go

```diff
@@ -104,9 +176,38 @@ func WithoutConstantNears(ctx context.Context, g *d2graph.Graph) (nears []*d2gra
 					break
 				}
 			}
+
+			obj.Parent = tempGraph.Root
```

Comment:
i'd prefer this to be moved up next to line 161, where ur setting tempGraph.Root.Children.

Reply:
line 172 still use `obj.Parent`

Reply:
ohh i see.